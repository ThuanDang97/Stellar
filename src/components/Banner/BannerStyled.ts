import styled from 'styled-components'

// Components
import { BoxShadowStyled } from '@components/styled-components'

// Utils
import { pxToRem } from '@utils/theme'

export const BannerStyled = styled(BoxShadowStyled)`
  position: relative;

  display: flex;
  justify-content: space-between;

  width: 100%;
  height: ${pxToRem(150)};
  padding: ${({ theme }) => pxToRem(theme.metrics.dimensions.md)};
  background-color: ${({ theme }) => theme.colors.white};
`

export const ImageWrapper = styled.div`
  position: absolute;
  right: ${pxToRem(-4)};

  bottom: ${({ theme }) => pxToRem(-theme.metrics.dimensions.lg)};

  width: ${({ theme }) => pxToRem(theme.metrics.images.banner)};
  height: ${({ theme }) => pxToRem(theme.metrics.images.banner)};

  @media (max-width: ${({ theme }) => pxToRem(theme.metrics.breakPoints.xs)}) {
    width: ${pxToRem(210)};
    height: ${pxToRem(210)};
    right: 0;
  }
`
