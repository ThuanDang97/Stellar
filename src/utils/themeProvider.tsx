import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
