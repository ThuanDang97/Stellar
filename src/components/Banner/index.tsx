import React, { memo, useEffect, useState } from 'react'
import { useTheme } from 'styled-components'

// Components
import Image from 'next/image'
import {
  TitleSecSionStyled,
  ViewStyled,
  TextDescriptionStyled,
} from '@components/styled-components'
import { BannerStyled, ImageWrapper } from '@components/Banner/BannerStyled'

// Props type
import { BannerProps } from '@self-types/components/Banner.props'

// Constants
import { BANNER_TITLE, BANNER_DESCRIPTION } from '@constants/index'

const Banner = ({ isLinkBill }: BannerProps) => {
  const theme = useTheme()
  const [innerWidth, setInnerWidth] = useState(window.screen.width)

  useEffect(() => {
    const handleResize = () => {
      setInnerWidth(window.screen.width)
    }
    // Add event  and remove event resize screen
    window.addEventListener('resize', handleResize)
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  const isMobile = innerWidth <= theme.metrics.breakPoints.xs
  return (
    <BannerStyled>
      {/* Text */}
      <ViewStyled
        width="55%"
        mBottom={isMobile ? theme.metrics.dimensions.base : 0}
      >
        <TitleSecSionStyled
          fontSize={theme.typography.fontSize.lg}
          fontFamily="AdobeCleanExtraBold"
          textAlign="left"
          mRight={theme.metrics.dimensions.xs}
        >
          {isLinkBill ? BANNER_TITLE.LINKED_BILL : BANNER_TITLE.NO_LINK_BILL}
        </TitleSecSionStyled>
        <Image
          src={isLinkBill ? '/icons/party.svg' : '/icons/flame.svg'}
          width={theme.metrics.dimensions.md}
          height={theme.metrics.dimensions.md}
          alt="flame"
        />

        <TextDescriptionStyled
          textAlign="left"
          as="p"
          mTop={theme.metrics.dimensions.sm}
          fontSize={theme.typography.fontSize.sm}
          nameComponent="banner"
        >
          {isLinkBill
            ? BANNER_DESCRIPTION.LINKED_BILL
            : BANNER_DESCRIPTION.NO_LINK_BILL}
        </TextDescriptionStyled>
      </ViewStyled>

      {/* Dog Image */}
      <ImageWrapper>
        <Image src="/images/space-dog.png" layout="fill" alt="space-dog" />
      </ImageWrapper>
    </BannerStyled>
  )
}

export default memo(Banner)
