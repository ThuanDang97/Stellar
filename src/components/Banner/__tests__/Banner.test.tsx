// Components
import Banner from '@components/Banner'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

describe('Banner Component', () => {
  it('renders Banner when not link bill unchanged', () => {
    const { asFragment } = renderWithTheme(<Banner isLinkBill={false} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('renders Banner after linked bill unchanged', () => {
    const { asFragment } = renderWithTheme(<Banner isLinkBill />)

    expect(asFragment()).toMatchSnapshot()
  })
})
