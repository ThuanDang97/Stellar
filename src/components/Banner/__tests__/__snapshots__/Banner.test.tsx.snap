// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Component renders Banner after linked bill unchanged 1`] = `
<DocumentFragment>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 BannerStyled-sc-587aj7-0 fTsvHV csavJt SFwKm"
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 kcXXQi"
      width="55%"
    >
      <span
        class="TextStyled-sc-9xia41-0 TextStyled__TitleSecSionStyled-sc-9xia41-1 jEbBXO fYtVag"
        font-family="AdobeCleanExtraBold"
        font-size="24"
      >
        Woohoo! You did it.
      </span>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="flame"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <p
        class="TextStyled-sc-9xia41-0 TextStyled__TextDescriptionStyled-sc-9xia41-2 gnCdZZ fGqNzw"
        font-size="16"
      >
        You got rid of the gravity holding down your credit score. Keep adding bills to boost your score.
      </p>
    </div>
    <div
      class="BannerStyled__ImageWrapper-sc-587aj7-1 eQlXBn"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="space-dog"
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Banner Component renders Banner when not link bill unchanged 1`] = `
<DocumentFragment>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 BannerStyled-sc-587aj7-0 fTsvHV csavJt SFwKm"
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 kcXXQi"
      width="55%"
    >
      <span
        class="TextStyled-sc-9xia41-0 TextStyled__TitleSecSionStyled-sc-9xia41-1 jEbBXO fYtVag"
        font-family="AdobeCleanExtraBold"
        font-size="24"
      >
        You’re Almost Done
      </span>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="flame"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <p
        class="TextStyled-sc-9xia41-0 TextStyled__TextDescriptionStyled-sc-9xia41-2 gnCdZZ fGqNzw"
        font-size="16"
      >
        Let’s set up your first bill and Stellar will make sure they get paid on time!
      </p>
    </div>
    <div
      class="BannerStyled__ImageWrapper-sc-587aj7-1 eQlXBn"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="space-dog"
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
  </div>
</DocumentFragment>
`;
